homeassistant/components/rotel/media_player.py

```diff
@@ -0,0 +1,493 @@
+"""
+Rotel platform.
+
+Although only the RSP-1570 is supported at the moment, the
+low level library could easily be updated to support other
+products of a similar vintage.
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    DOMAIN, PLATFORM_SCHEMA, MediaPlayerDevice)
+from homeassistant.components.media_player.const import (
+    SUPPORT_SELECT_SOURCE, SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_DEVICE, CONF_NAME, EVENT_HOMEASSISTANT_STOP,
+    STATE_OFF, STATE_ON)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Rotel RSP-1570'
+
+SUPPORT_ROTEL_RSP1570 = \
+    SUPPORT_VOLUME_SET | SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+ROTEL_RSP1570_SOURCES = {
+    ' CD': 'SOURCE_CD',
+    'TUNER': 'SOURCE_TUNER',
+    'TAPE': 'SOURCE_TAPE',
+    'VIDEO 1': 'SOURCE_VIDEO_1',
+    'VIDEO 2': 'SOURCE_VIDEO_2',
+    'VIDEO 3': 'SOURCE_VIDEO_3',
+    'VIDEO 4': 'SOURCE_VIDEO_4',
+    'VIDEO 5': 'SOURCE_VIDEO_5',
+    'MULTI': 'SOURCE_MULTI_INPUT',
+}
+
+CONF_SOURCE_ALIASES = "source_aliases"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_DEVICE): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
```

Comment:
Entity names can be edited and overridden in the front-end, this isn't necessary for config, so I suggest removing it and just using the default in the class directly.

Reply:
Thanks for the thorough review.    I'm working on the component implementation as we speak.

I was a little confused by the comment regarding not setting CONF_NAME through config though.   I can't see how a media player entity friendly name can be set from the front end.    The documentation says that the friendly name can be edited for "supported entities" - is this supported for the media player?

I've been looking for a good example of an entity that doesn't require CONF_NAME to be set but the only ones that don't have logic similar to mine seem to be able to pull the name from the device itself.

Could you elaborate a little on what I should do please?

Reply:
If the entity has a unique_id property set, the name and entity_id can va changed in the frontend.